common_utils.py

```
@@ -11,6 +16,13 @@
 random.seed(42) # for reproducibility
 
 # OUR LLM + EMBEDDING SETUP
-set_seed(42)  # for reproducibility
-generator = pipeline("text-generation", model="gpt2")
-embedder = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2")
+
+def get_generator(name: str, *args, **kwargs) -> AbstractGenerator:
+    if name == "gpt2":
+        return GPT2Generator(*args, **kwargs)
+    return None
+
+def get_encoder(name: str, *args, **kwargs) -> AbstractEncoder:
+    if name == "MiniLM":
+        return MiniLMEncoder(*args, **kwargs)
+    return None
```

Same about this line, I would rather throw an exception here.